15_mini_project_work/2_num_guessing_game.py

```diff
@@ -37,31 +39,37 @@ def is_valid(input_string):
 def main():
     print(GREETING)
 
-    is_guess_wrong = True
-    hidden_num = random.randint(RANDOM_LEFT_BORDER, RANDOM_RIGHT_BORDER)
-    guess_count = 1
-
-    while is_guess_wrong:
-        guess = input(PROMPT_MESSAGE)
-        if not is_valid(guess):
-            print(ERROR_MESSAGE)
-        else:
-            guess_num = int(guess)
-            if guess_num < hidden_num:
-                print(TOO_SMALL_MESSAGE)
-            elif guess_num > hidden_num:
-                print(TOO_BIG_MESSAGE)
-            else:
-                print(WIN_MESSAGE)
-                print(GUESS_NUMBER_MESSAGE, guess_count, sep='\n')
-                print(FAREWELL_MESSAGE)
+    game_is_going_on = True
+    while game_is_going_on:
+
+        hidden_num = random.randint(RANDOM_LEFT_BORDER, RANDOM_RIGHT_BORDER)
+        guess_count = 1
 
-                is_guess_wrong = False
-        guess_count += 1
+        is_guess_wrong = True
+        while is_guess_wrong:
+            guess = input(PROMPT_MESSAGE)
 
+            if not is_valid(guess):
+                print(ERROR_MESSAGE)
+            else:
+                guess_num = int(guess)
+                if guess_num < hidden_num:
+                    print(TOO_SMALL_MESSAGE)
+                elif guess_num > hidden_num:
+                    print(TOO_BIG_MESSAGE)
+                else:
+                    print(WIN_MESSAGE)
+                    is_guess_wrong = False
+            guess_count += 1
+        new_game_wish = input(NEW_GAME_PROPOSAL_MESSAGE).lower()
+        if new_game_wish != YES_RESPONSE:
+            print(GUESS_NUMBER_MESSAGE, guess_count, sep='\n')
+            print(FAREWELL_MESSAGE)
```

Comment:
By logic - it should be the VERY last action of our game.
Hence, let's move it out of loop and make the last statement

Reply:
Done